### DISCUSSION TITLE
NoClassDefFoundError thrown in QuarkusCxfProcessor generateClasses() during build.

### DISCUSSION THREAD

**Discussion by Yaroon** [quarkiverse/quarkus-cxf#1302] [Q&A] [answered]
I am trying to implement a SOAP service through generated code. The code generation went well and code is generated in 3 different packages in generated-sources/wsdl2java.

In the build step however, the following exception occurs:

```
 Failed to execute goal io.quarkus.platform:quarkus-maven-plugin:3.8.1:build (default) on project adapter-ninas: Failed to build quarkus application: io.quarkus.builder.BuildException: Build failure: Build failed due to errors
[ERROR] 	[error]: Build step io.quarkiverse.cxf.deployment.QuarkusCxfProcessor#generateClasses threw an exception: java.lang.NoClassDefFoundError: java/be_telenet_comm_soa/BusinessException
```

I am wondering what I would have to do in order for Quarkus to find the BusinessException in the java.be_telenet_comm_soa package.

[BusinessException.java.zip](https://github.com/quarkiverse/quarkus-cxf/files/14651017/BusinessException.java.zip)
[BusinessException.java.zip](https://github.com/quarkiverse/quarkus-cxf/files/14651028/BusinessException.java.zip)

**Comment by Yaroon** [comment]
I might come up with a small project to reproduce the problem, but the problem is in the java:be.telenet.com.soa.exception namespace for an element. That is an "unorthodox" namespace, even though anything goes for a namespace... it does not have to be a URI. It usually is a URI though (resolvable or not) In any case, if it is not a normal URI it seems that something goes wrong in the generated packages during the WSDL2Java code generation.

**Reply by ppalaga** [reply, marked as answer]
I would say `wsdl2java` is generating the classes in the way it is instructed to, or better: in the way it usually transforms namespaces to class names.

I think the problem is that some classes are generated under `java` Java package, which is reserved for JRE classes. IIRC no classloader is allowed to load classes from `java.*` packages except for Bootstrap Class Loader. And that one in turn will not load any classes from user jars. Hence the class is not visible to any class loader. 

You may want to try to map your peculiar namespace uri to a different Java package using `quarkus.cxf.codegen.wsdl2java.package-names`: 
![image](https://github.com/quarkiverse/quarkus-cxf/assets/1826249/70013600-eafb-4535-965b-41cf803d14fa)
